api/types/database.go

```diff
@@ -330,6 +329,22 @@ func (d *DatabaseV3) Copy() *DatabaseV3 {
 	return proto.Clone(d).(*DatabaseV3)
 }
 
+// MatchSearch goes through select field values and tries to
+// match against the list of search values.
+func (d *DatabaseV3) MatchSearch(values []string) bool {
+	fieldVals := append(utils.MapToStrings(d.GetAllLabels()), d.GetName(), d.GetDescription(), d.GetProtocol(), d.GetType())
+
+	var custom func(string) bool
+	switch d.GetType() {
+	case DatabaseTypeCloudSQL:
+		custom = func(val string) bool {
+			return strings.EqualFold(val, "cloud") || strings.EqualFold(val, "cloud sql")
```

Comment:
Why do we have a special case for Cloud SQL here and for `"tunnel"` for SSH servers?

https://github.com/gravitational/teleport/blob/513f874fab01a8ea89c6edf0ac6f47501fd0553b/api/types/server.go#L427-L431

I'm trying to understand the current behavior because I'll have to more or less temporarily mimic `MatchSearch` behavior in Connect to score the search results according to how well they match the given search terms.

Reply:
For `tunnel`, it was to handle the case where users typed a search term `tunnel`, then we give back the results that use tunnel. Before server side filtering, we had client side filtering, and i wanted to mimic the same searching experience.

The value for `DatabaseTypeCloudSQL` is `gcp`, and i figured users may search for gcp related stuff by typing `cloud sql` instead of `gcp`.

Reply:
Thanks for the explanation. I was looking at `MatchSearch` of `DatabaseServiceV1` and I thought we're only matching on db name and labels, I totally missed that `DatabaseV3` has the `MatchSearch` method as well.

I didn't check how the situation looked like when this PR was merged, but nowadays AFAIK both the Web UI and Connect split the search terms by space, so in that case val will never be equal to "cloud sql". But that's just something I'll have to remember about in the scoring algorithm.